### DISCUSSION TITLE
cluster_pairwise_predictions_at_threshold() failing when linking two 140M records

### DISCUSSION THREAD

**Discussion by reyvasquez-vh** [moj-analytical-services/splink#1037] [Q&A] [answered]
Previously, I had issues with linking two 35M records.  See https://github.com/moj-analytical-services/splink/discussions/1017
That works now.  Thanks!

I am now trying to link two 140M records, and the code fails at `linker.cluster_pairwise_predictions_at_threshold()`
The execution printouts a lot of "No more replicas available" 

And also:
A lot of `Failed to send RPC RPC 5462906116369655171 to /<ApplicationMaster IP>:35100:` error

**Comment by reyvasquez-vh** [comment, marked as answer]
The problem here was with storage for checkpoints.  Pointing checkpoint to S3 bucket instead of using node spaces fixed the issue.

**Reply by RobinL** [reply]
thanks - appreciate you posting the solution
